cloud/scope/load_balancer_reconciler.go

```diff
@@ -37,6 +37,9 @@ func (s *ClusterScope) ReconcileApiServerLB(ctx context.Context) error {
 		return err
 	}
 	if lb != nil {
+		if lb.LifecycleState != loadbalancer.LoadBalancerLifecycleStateActive {
+			return errors.New("load balancer is not in active state, hence reconciliation cannot happen")
```

Comment:
What about something like 

```suggestion
			return errors.New(fmt.Sprintf("load balancer is not in %s. Waiting for active state.", lb.LifecycleState))
```

or something. 🤷 This way we tell the user we aren't moving forward, but also let them know the current state.

Also what happens if the LB is in a failed state? Will we sit here forever waiting?

Reply:
fixed, please check..

Reply:
yes, in case of failed state, or in any non active state, we will go into exponential backoff will the operator goes off and investigates. We would have tried recreating the LB but that would have been more code changes and I did not want take that risk at this point.